File: flutter/fn.giwu/lib/api/bookmarks.dart
import '../models/bookmark.dart';
import 'client.dart';

Future<List<BookmarkModel>> listBookmarks() async {
  final response = await dio.get('/bookmarks');
  final list = response.data as List<dynamic>;
  return list
      .map((e) => BookmarkModel.fromJson(e as Map<String, dynamic>))
      .toList();
}

Future<BookmarkModel> saveBookmark({
  required String bible,
  required int book,
  required int chapter,
  required int verse,
  required String text,
}) async {
  final response = await dio.post('/bookmarks', data: {
    'bible': bible,
    'book': book,
    'chapter': chapter,
    'verse': verse,
    'text': text,
  });
  return BookmarkModel.fromJson(response.data as Map<String, dynamic>);
}

Future<void> deleteBookmark(int id) async {
  await dio.delete('/bookmarks/$id');
}

File: js/fn.giwu/src/api/bookmarks.js
import client from './client'

export async function listBookmarks() {
  const { data } = await client.get('/bookmarks')
  return data
}

export async function saveBookmark(bible, book, chapter, verse, text) {
  const { data } = await client.post('/bookmarks', { bible, book, chapter, verse, text })
  return data
}

export async function deleteBookmark(id) {
  await client.delete(`/bookmarks/${id}`)
}
